File: cloud/dao/system_notice.go
package dao

import "github.com/abulo/ratel/v3/stores/null"

// SystemNotice 通知公告表 system_notice
type SystemNotice struct {
	Id         *int64        `db:"id" json:"id"`                  //bigint 公告ID,PRI
	Title      *string       `db:"title" json:"title"`            //varchar 公告标题
	Content    *string       `db:"content" json:"content"`        //text 公告内容
	Type       *int32        `db:"type" json:"type"`              //tinyint 公告类型（1通知 2公告）
	Status     *int32        `db:"status" json:"status"`          //tinyint 公告状态（0正常 1关闭）
	Deleted    *int32        `db:"deleted" json:"deleted"`        //tinyint 删除
	TenantId   *int64        `db:"tenant_id" json:"tenantId"`     //bigint 租户
	Creator    null.String   `db:"creator" json:"creator"`        //varchar 创建人
	CreateTime null.DateTime `db:"create_time" json:"createTime"` //datetime 创建时间
	Updater    null.String   `db:"updater" json:"updater"`        //varchar 更新人
	UpdateTime null.DateTime `db:"update_time" json:"updateTime"` //datetime 更新时间
}

File: admin/src/api/interface/systemNotice.ts
// system_notice 通知公告表
import { ReqPage } from "./index";
export namespace SystemNotice {
  export interface ReqSystemNoticeList extends ReqPage {
    deleted?: number; // 删除
    status?: number; // 公告状态（0正常 1关闭）
    type?: number; // 公告类型（1通知 2公告）
    title?: string; // 公告标题
  }
  export interface ResSystemNoticeItem {
    id: number; // 公告ID
    title: string; // 公告标题
    content: string; // 公告内容
    type: number; // 公告类型（1通知 2公告）
    status: number; // 公告状态（0正常 1关闭）
    deleted: number; // 删除
    tenantId: number; // 租户
    creator: string | undefined; // 创建人
    createTime: string | undefined; // 创建时间
    updater: string | undefined; // 更新人
    updateTime: string | undefined; // 更新时间
  }
}
